kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -199,6 +195,29 @@ public T get() {
     }
   }
 
+  @Override
+  public T require() throws ResourceNotFoundException {
+    try {
+      T answer = getMandatory();
+      if (answer == null) {
+        throw new ResourceNotFoundException("The resource you request doesn't exist or couldn't be fetched.");
+      }
+      if (answer instanceof HasMetadata) {
+        HasMetadata hasMetadata = (HasMetadata) answer;
+        updateApiVersion(hasMetadata);
+      } else if (answer instanceof KubernetesResourceList) {
+        KubernetesResourceList list = (KubernetesResourceList) answer;
+        updateApiVersion(list);
+      }
+      return answer;
+    } catch (KubernetesClientException e) {
+      if (e.getCode() != 404) {
```

Comment:
just a suggestion, better replace this by `HttpURLConnection.HTTP_NOT_FOUND` here and at some other places in this file